infra/build/functions/sync/main.py

```diff
@@ -35,41 +47,141 @@ class GitAuth(ndb.Model):
   access_token = ndb.StringProperty()
 
 
-def sync_projects(projects):
+def create_scheduler(cloud_scheduler_client, project_name, schedule):
+  """Creates schedulers for new projects."""
+  project_id = os.environ.get('GCP_PROJECT')
+  location_id = os.environ.get('FUNCTION_REGION')
+  parent = cloud_scheduler_client.location_path(project_id, location_id)
+  job = {
+      'name': parent + '/jobs/' + project_name + '-scheduler',
+      'pubsub_target': {
+          'topic_name': 'projects/' + project_id + '/topics/request-build',
+          'data': project_name.encode()
+      },
+      'schedule': schedule
+  }
+
+  cloud_scheduler_client.create_job(parent, job)
+
+
+def delete_scheduler(cloud_scheduler_client, project_name):
+  """Deletes schedulers for projects that were removed."""
+  project_id = os.environ.get('GCP_PROJECT')
+  location_id = os.environ.get('FUNCTION_REGION')
+  name = cloud_scheduler_client.job_path(project_id, location_id,
+                                         project_name + '-scheduler')
+  cloud_scheduler_client.delete_job(name)
+
+
+def update_scheduler(cloud_scheduler_client, project, schedule):
+  """Updates schedule in case schedule was changed."""
+  project_id = os.environ.get('GCP_PROJECT')
+  location_id = os.environ.get('FUNCTION_REGION')
+  parent = cloud_scheduler_client.location_path(project_id, location_id)
+  job = {
+      'name': parent + '/jobs/' + project.name + '-scheduler',
+      'pubsub_target': {
+          'topic_name': 'projects/' + project_id + '/topics/request-build',
+          'data': project.name.encode()
+      },
+      'schedule': project.schedule
+  }
+
+  update_mask = {'schedule': schedule}
+  cloud_scheduler_client.update(job, update_mask)
+
+
+def sync_projects(cloud_scheduler_client, projects):
   """Sync projects with cloud datastore."""
   project_query = Project.query()
```

Comment:
no need for project_query variable. Just use Project.query() directly where it's used.

Reply:
https://github.com/google/oss-fuzz/pull/4023#discussion_r444613188

This was changed to be this way in the initial cloud function integration PR since it is queried multiple times, should i change it back?

Reply:
Ah, you misunderstood my initial comment. My initial comment was to save the results of `list(Project.query())` (note the `list`) so that the query is only run once.  Saving the Query object into a variable (without iterating through it) does not ensure that it only runs once.

With the current state where we are modifying existing projects etc, doing this is no longer simple, so doing `list(Project.query())` is no longer ideal.